README.md

```diff
@@ -10,33 +10,58 @@ Telepoison is a [opentelemetry-instrumented](https://github.com/open-telemetry/o
 
 ## Usage
 
-Simply replace `HTTPoison` with `Telepoison` when calling one of the request methods (get(), get!(), post(), request(), etc.)
+Call `Telepoison.setup/1` within your application start-up.
+
+Replace usages of the `HTTPoison` module with `Telepoison` when calling one of the *derived* request functions provided by `HTTPoison` (`HTTPoison.get/3`, `HTTPoison.get!/3` etc.)
 
 ```elixir
-# before
+# Before
 HTTPoison.get!(url, headers, opts)
 
-# after
+# After
 Telepoison.get!(url, headers, opts)
 ```
 
-Additionally, telepoison adds some options that can be passed in the `opts` HTTPoison argument to set OpenTelemetry-related stuff.
-These start with `:ot_`
+## Configuration
+
+`Telepoison.setup/1` takes a `Keyword list` that can configure if and how the `http.route` Open Telemetry metadata will be set per request using the `:infer_route` option.
+
+* If no value is provided then the out of the box, conservative inference provided by `Telepoison.URI.infer_route_from_request/1` is used to determine the inference
+
+* If an function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) is provided then that function is used to determine the inference
+
+This can be overridden per each call to `Telepoison.request/1` derived functions (`Telepoison.get/3`, `Telepoison.get!/3` etc.)
+
+## Open Telemetry integration
 
-* `:ot_span_name` sets the span name
-* `:ot_attributes` additional span attributes that will be added to the span. Should be a list of {name, value} tuples.
+Additionally, `Telepoison` provides some options that can be added to each derived function via
+the `Keyword list` `opts` parameter (or the `HTTPoison.Request` `options` `Keyword list` if calling `Telepoison.Request/1` directly). These are prefixed with `:ot_`.
 
+* `:ot_span_name` - sets the span name.
+* `:ot_attributes` - a list of `{name, value}` `tuple` attributes that will be added to the span.
+* `:ot_resource_route` - sets the `http.route` attribute, depending on the value provided.
+
+If the value is a string or an function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) that is used to set the attribute
+
+If `:infer` is provided, then the function discussed within the *Configuration* section is used to set the attribute
```

Comment:
```suggestion
If `:infer` is provided, then the function discussed within the (*Configuration*)[#configuration] section is used to set the attribute
```

I don't know if this is the exact syntax, but I'd add a link to the section if possible

Reply:
Updated!